eliot-service/eliot/symbolicate_resource.py

```diff
@@ -380,137 +380,143 @@ def get_symcache(self, module_info, debug_stats):
-                            sourceloc = sourceloc_list[-1]
-
-                            data["function"] = sourceloc.symbol
-                            data["function_offset"] = hex(
-                                module_offset - sourceloc.sym_addr
-                            )
-                            if sourceloc.full_path:
-                                data["file"] = sourceloc.full_path
-
-                            # Only add a "line" if it's non-zero and not None, and if there's a
-                            # file--otherwise the line doesn't mean anything
-                            if sourceloc.line and data.get("file"):
-                                data["line"] = sourceloc.line
-
-                            if len(sourceloc_list) > 1:
-                                # We have inline information. Add an "inlines" property with a list
-                                # of { function, file, line } entries.
-                                inlines = []
-                                for inline_sourceloc in sourceloc_list[:-1]:
-                                    inline_data = {
-                                        "function": inline_sourceloc.symbol,
-                                    }
-
-                                    if inline_sourceloc.full_path:
-                                        inline_data["file"] = inline_sourceloc.full_path
-
-                                    if inline_sourceloc.line and inline_data.get(
-                                        "file"
-                                    ):
-                                        inline_data["line"] = inline_sourceloc.line
-
-                                    inlines.append(inline_data)
-
-                                data["inlines"] = inlines
-
-                    data["module"] = module_info.filename
-
-                symbolicated_stack.append(data)
-
-            symbolicated_stacks.append(symbolicated_stack)
-
-        # Convert modules to a map of debug_filename/debug_id -> True/False/None
-        # on whether we found the sym file (True), didn't find it (False), or never
-        # looked for it (None)
-        found_modules = {
-            f"{module_info.debug_filename}/{module_info.debug_id}": module_info.has_symcache
-            for module_info in module_records
-        }
+        results = []
+        for i, job in enumerate(jobs):
+            stacks = job.get("stacks", [])
+            modules = job.get("memoryMap", [])
+
+            # Build list of Module instances so we can keep track of what we've
+            # used/seen
+            module_records = [
+                ModuleInfo(
+                    filename=debug_filename,
+                    debug_filename=debug_filename,
+                    debug_id=debug_id,
+                    has_symcache=None,
+                    symcache=None,
+                )
+                for debug_filename, debug_id in modules
+            ]
```

Comment:
To reduce work acquiring symcache files, we should bump this outside of the `for` loop. However, right now we're having memory errors so having fewer modules in memory at the same time is better. However, however, we're not seeing any multi-job symbolication requests right now so it doesn't matter.

Reply:
> However, however, we're not seeing any multi-job symbolication requests right now so it doesn't matter.

The profiler uses multi-job requests, but it only requests a single module per job.
https://github.com/firefox-devtools/profiler/blob/b5f7ff53dba06e4918eafec88a2d17bb41239bfc/src/profile-logic/mozilla-symbolication-api.js#L276-L287

Reply:
Huh. My metrics show no indication of multiple-jobs-per-request. I'll have to look into that.